osgar/tools/lidarview.py

```diff
@@ -2,35 +2,67 @@
   Draw laser scans
 """
 import math
+import io
 from datetime import timedelta
 from ast import literal_eval
 
 import pygame
 from pygame.locals import *
 
-from osgar.logger import LogReader, lookup_stream_id
+from osgar.logger import LogReader, lookup_stream_id, lookup_stream_names
 from osgar.lib.serialize import deserialize
 
 
 WINDOW_SIZE = 1200, 660
+TAIL_MIN_STEP = 0.1  # in meters
+g_scale = 30
 
 
-def scans_gen(logfile, stream_id):
-    only_stream = lookup_stream_id(logfile, stream_id)
+def scans_gen(logfile, lidar_name, poses_name=None, camera_name=None):
+    """
+    Generator for (timestamp, pose, lidar, image) where freqency is defined
+    by LIDAR and pose and image is used the most recent
+    """
+    lidar_id, poses_id, camera_id = None, None, None
+
+    if poses_name is None:
+        only_stream = lookup_stream_id(logfile, lidar_name)
+        lidar_id = only_stream
+    else:
+        names = lookup_stream_names(logfile)
+        lidar_id = names.index(lidar_name) + 1
+        poses_id = names.index(poses_name) + 1
+        only_stream = [lidar_id, poses_id]
+        if camera_name is not None:
+            camera_id = names.index(camera_name) + 1
+            only_stream.append(camera_id)
```

Comment:
It is just a small nit, but this if-else part could probably be improved. It is not immediately obvious which combination of inputs is supported. It seems that it is not possible to include `camera_name` if `pose_name` is None, right? Can we have two separate ifs that independently append to `only_stream` it the name is set?

```python
only_stream = [ lookup_stream_id(logfile, lidar_name) ]
if poses_name is not None:
  only_stream.append( lookup_stream_id(logfile, poses_name) )
if camera_name is not None:
  only_stream.append( lookup_stream_id(logfile, camera_name) )
```
?

Reply:
Thanks - I was not happy with this section too. I plan to make all inputs optional and sampling given by lidar, camera, poses in this order. If none is specified then it should list all names.